src/ElfStringTable.c

```diff
@@ -7,9 +7,9 @@ char* getSectionString(ElfImageP elfI, Elf32_Word sectionNo)
   return (char*)&elfI->strTable.secStrs[elfI->sections.tab[sectionNo].sh_name];
 }
 
-char* getSymbolString(ElfImageP elfI, Elf32_Word offIntoTab)
+char* getSymbolString(ElfImageP elfI, Elf32_Word symbolNo)
 {
-  return (char*)&elfI->strTable.symStrs[offIntoTab];
+  return (char*)&elfI->strTable.symStrs[elfI->symbols.tab[symbolNo].st_name];
```

Comment:
st_name is an offset into a string table (shstrtab or strtab)
it could be in symStrs or secStrs
to know which one to pick see st_info with ELF32_ST_TYPE
reread: docs/Format_ELF.pdf
FIXIT